src/Rector/v10/v3/UsePSR14EventsExtbaseRelatedSignalsRector.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Ssch\TYPO3Rector\Rector\v10\v3;
+
+use PhpParser\Node;
+use PhpParser\Node\Arg;
+use PhpParser\Node\Expr\MethodCall;
+use Rector\Core\Rector\AbstractRector;
+use Rector\Core\RectorDefinition\CodeSample;
+use Rector\Core\RectorDefinition\RectorDefinition;
+use TYPO3\CMS\Extbase\Mvc\Controller\ActionController;
+
+/**
+ * @see https://docs.typo3.org/c/typo3/cms-core/master/en-us/Changelog/10.3/Deprecation-89870-NewPSR-14EventsForExtbase-relatedSignals.html
+ */
+final class UsePSR14EventsExtbaseRelatedSignalsRector extends AbstractRector
+{
+    /**
+     * @return string[]
+     */
+    public function getNodeTypes(): array
+    {
+        return [MethodCall::class];
+    }
+
+    /**
+     * @param MethodCall $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        if (! $this->isMethodStaticCallOrClassMethodObjectType($node, ActionController::class)) {
+            return null;
+        }
+
+        if (null === $node->name) {
+            return null;
+        }
+
+        if ($this->isName($node->name, 'emitBeforeCallActionMethodSignal')) {
+            if (! (property_exists($node, 'args') && null !== $node->args)) {
+                return null;
+            }
+
+            /** @var Arg[] $args */
+            $args = $node->args;
+            $firstArgument = array_shift($args);
+
+            if (null === $firstArgument) {
+                return null;
+            }
+
+            $argumentValue = $this->getValue($firstArgument->value);
+
+            if (null === $argumentValue) {
+                return null;
+            }
+
+            return $this->createMethodCall($this->createPropertyFetch($node->var, 'eventDispatcher'), 'dispatch',
```

Comment:
Hi @sabbelasichon,

It's still work in progress, but I'm currently stucked here. Hope you can give me some pointers. As you see I end up with a `$new->BeforeActionCallEvent` instead of `new BeforeActionCallEvent` + the `static::class` shouldn't be a string.

I'm pretty sure it's fairly easy to fix, but I don't find the function needed for this. 

```php
     public function method(): void
     {
         $actionController = GeneralUtility::makeInstance(ActionController::class);
-        $actionController->eventDispatcher->dispatch(new BeforeActionCallEvent(static::class, 'indexAction', []));
+        $actionController->eventDispatcher->dispatch($new->BeforeActionCallEvent('static::class', 'indexAction', []));
     }
```

Reply:
Currently playing around with the `New_`

Reply:
This is my best shot so far. 

``` 
$this->createFuncCall( 'new BeforeActionCallEvent', ['static::class', 'indexAction', $firstArgument]),
```
But I'm pretty sure it's not correct with the "new BeforeActionCallEvent" as I'm still missing the `use statement` the `static::class` is still WIP. Didn't find a solution for that yet either.